controller/space.go

```diff
@@ -179,17 +232,169 @@ func (c *SpaceController) Delete(ctx *app.DeleteSpaceContext) error {
 		}
 		return appl.Spaces().Delete(ctx.Context, ctx.SpaceID)
 	})
-
 	if err != nil {
 		return jsonapi.JSONErrorResponse(ctx, err)
 	}
 	err = c.resourceManager.DeleteSpace(ctx, ctx.Request, ctx.SpaceID.String())
 	if err != nil {
 		return jsonapi.JSONErrorResponse(ctx, err)
 	}
+
+	// now delete the OpenShift resources associated with this space on an
+	// OpenShift cluster
+	err = deleteOpenShiftResource(c.DeploymentsClient, config, ctx.Context, spaceID)
+	if err != nil {
+		log.Error(ctx.Context, map[string]interface{}{
+			"space_id": spaceID,
+			"error":    err,
+		}, "could not delete OpenShift resources")
+		return jsonapi.JSONErrorResponse(
+			ctx, goa.ErrInternal(spaceDeletionErrorExternal))
+	}
+
 	return ctx.OK([]byte{})
 }
 
+// deleteCodebases deletes all the codebases that are associated with this space
+func deleteCodebases(
+	httpClient *http.Client,
+	config *configuration.Registry,
+	ctx context.Context,
+	spaceID goauuid.UUID) error {
+
+	u, err := url.Parse(config.GetCodebaseServiceURL())
+	if err != nil {
+		return errs.Wrapf(err, "malformed codebase service URL: %s", config.GetCodebaseServiceURL())
```

Comment:
if you intend to consider this an InternalError in the layers above this method, you could use `errors.NewInternalError...`

Reply:
done